pkg/pb/synthetic_monitoring/checks_extra.go

```diff
@@ -553,6 +610,253 @@ func (s *K6Settings) Validate() error {
 	return nil
 }
 
+func (s *MultiHttpSettings) Validate() error {
+	if len(s.Entries) == 0 {
+		return ErrInvalidMultiHttpTargets
+	}
+
+	if err := validateCollection(s.Entries); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *MultiHttpEntry) Validate() error {
+	if err := e.Request.Validate(); err != nil {
+		return err
+	}
+
+	if err := validateCollection(e.Assertions); err != nil {
+		return err
+	}
+
+	if err := validateCollection(e.Variables); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h HttpHeader) Validate() error {
+	if !httpguts.ValidHeaderFieldName(h.Name) {
+		return ErrInvalidHttpHeaders
+	}
+
+	if !httpguts.ValidHeaderFieldValue(h.Value) {
+		return ErrInvalidHttpHeaders
+	}
+
+	return nil
+}
+
+func (f QueryField) Validate() error {
+	if len(f.Name) == 0 {
+		return ErrInvalidQueryFieldName
+	}
+
+	// the value might be empty
+
+	// The name can be anything. TODO(mem): is this true?
+
+	return nil
+}
+
+func (r *MultiHttpEntryRequest) Validate() error {
+	if r == nil {
+		return nil
+	}
+
+	if err := r.Method.Validate(); err != nil {
+		return err
+	}
+
+	if err := validateHttpUrl(r.Url); err != nil {
+		return err
+	}
+
+	// TODO(mem): do something with HttpVersion?
+
+	if err := r.Body.Validate(); err != nil {
+		return err
+	}
+
+	if err := validateCollection(r.Headers); err != nil {
+		return err
+	}
+
+	if err := validateCollection(r.QueryFields); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validate verifies that the MultiHttpEntryAssertion is valid.
+//
+// Because of the structure represents multiple orthogonal variants, this
+// function has to branch based on the type.
+//
+//nolint:gocyclo
+func (a *MultiHttpEntryAssertion) Validate() error {
```

Comment:
Nice! I was thinking that this part was going to be painful in go, interesting to see how that works

Reply:
I had to try a couple of times :-P

I'm not 100% happy with _not_ validating the expressions (e.g. json path or CSS selector) but I went and looked and those are implemented in javascript, not Go, so I don't want to deal with the headache of having to different implementations with different bugs.